QUARKUS-6521.MD

```diff
@@ -0,0 +1,57 @@
+# QUARKUS-6521 - Virtual thread support for smallrye-graphql
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-6521
+
+Upstream PR: https://github.com/quarkusio/quarkus/pull/47802
+
+Quarkus 3.25 adds support for running GraphQL endpoints on virtual threads.
+Previously, endpoint callbacks could be executed either on an IO thread or a worker thread.
+This new feature was a minor update to the GraphQL extension.
+When the GraphQL extension executes blocking code, it checks whether the endpoint is annotated with the RunOnVirtualThread annotation and selects the virtual thread executor instead of the worker thread executor.
+
+## Scope of the testing
+
+Automated test coverage should verify that:
+
+* All Quarkus GraphQL endpoints with a synchronous return type or those annotated @Blocking can be successfully migrated to and executed on virtual threads.
+* GraphQL endpoint code does not pin the carrier thread.
+* Security, telemetry, and GraphQL clients integration works with this  new feature.
+* PostgreSQL JDBC driver can be used to access the database from virtual threads. Quarkus documentation warns that most JDBC drivers still pin the carrier thread. With the [JEP 491](https://openjdk.org/jeps/491) and Java 25, the situation is likely to improve. Until then, our goal is to verify that users can at least access the database from virtual threads.
+* OpenShift scenarios, JVM mode, DEV mode, and native mode scenarios.
+
+Manual verification should assert low memory consumption and application performance under heavy load.
+
+## Existing test coverage
+
+There is no virtual threads test coverage in any of our Quarkus QE test suites.
+The one test that uses the RunOnVirtualThread annotation is a copy-and-paste of the original reproducer, and the original issue wasn’t caused by using virtual threads.
+
+Upstream test coverage verifies that GraphQL endpoints annotated with the RunOnVirtualThread annotation are executed on virtual threads, and the carrier thread is not pinned.
+The detection of pinning cases relies on JFR support.
+The synchronization block, relevant for Java 21, 22, and 23, is also tested.
```

Comment:
We also run testing on Java 17, so do not forget to disable the tests there, please (you may include this note to the tests plans if you want.)

Reply:
Thanks for the reminder, I have added note to `Impact on test suites and testing automation` that tests will run with Java 21 and higher.